src/document/mod.rs

```diff
@@ -84,113 +116,225 @@ impl DocumentEncryptOpts {
+    /// Returns the metadata for an IronCore encrypted document.
     ///
     /// # Arguments
     /// - `id` - Unique ID of the document to retrieve
     ///
-    /// # Returns
-    /// `DocumentMetadataResult` with details about the requested document.
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let document_id: DocumentId = unimplemented!();
+    /// let document_data = sdk.document_get_metadata(&document_id).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
     async fn document_get_metadata(&self, id: &DocumentId) -> Result<DocumentMetadataResult>;
 
-    /// Attempt to parse the document ID out of an encrypted document.
+    /// Returns the document ID from the bytes of an IronCore encrypted document.
+    ///
+    /// This is the same ID returned by `DocumentEncryptResult.id()`.
     ///
     /// # Arguments
-    /// - `encrypted_document` - Encrypted document bytes
+    /// - `encrypted_document` - Bytes of the encrypted document
+    ///
+    /// # Errors
+    /// Fails if the provided encrypted bytes have no header.
     ///
-    /// # Returns
-    /// `Result<DocumentId>` Fails if provided encrypted document has no header, otherwise returns extracted ID.
+    /// # Examples
+    /// ```
+    /// # fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let bytes: Vec<u8> = vec![];
+    /// let document_id = sdk.document_get_id_from_bytes(&bytes)?;
+    /// # Ok(())
+    /// # }
+    /// ```
     fn document_get_id_from_bytes(&self, encrypted_document: &[u8]) -> Result<DocumentId>;
 
-    /// Encrypt the provided document bytes.
+    /// Updates the contents of an existing IronCore encrypted document.
     ///
-    /// # Arguments
-    /// - `document_data` - Bytes of the document to encrypt
-    /// - `encrypt_opts` - Optional document encrypt parameters. Includes
-    ///       `id` - Unique ID to use for the document. Document ID will be stored unencrypted and must be unique per segment.
-    ///       `name` - Non-unique name to use in the document. Document name will **not** be encrypted.
-    ///       `grant_to_author` - Flag determining whether to encrypt to the calling user or not. If set to false at least one value must be present in the `grant` list.
-    ///       `grants` - List of users/groups to grant access to this document once encrypted
-    async fn document_encrypt(
-        &self,
-        document_data: &[u8],
-        encrypt_opts: &DocumentEncryptOpts,
-    ) -> Result<DocumentEncryptResult>;
-
-    /// Update the encrypted content of an existing document. Persists any existing access to other users and groups.
+    /// The new contents will be encrypted, and which users and groups are granted access
+    /// will remain unchanged.
```

Comment:
Removing as this has been combined with the above.
```suggestion
```

Reply:
See above